internal/cli/commands.go

```diff
@@ -3,110 +3,146 @@ package cli
-		return dir
+func timeVersion(startTime time.Time, format string) (version string, err error) {
+	switch format {
+	case "":
+		err = errInvalidTimeFormat
+	case "unix":
+		version = strconv.FormatInt(startTime.Unix(), 10)
+	case "unixNano":
+		version = strconv.FormatInt(startTime.UnixNano(), 10)
 	default:
-		return dir + "/"
+		version = startTime.Format(format)
 	}
+
+	return
 }
 
 // createCmd (meant to be called via a CLI command) creates a new migration
-func createCmd(dir string, startTime time.Time, format string, name string, ext string, seq bool, seqDigits int) {
-	dir = cleanDir(dir)
-	var base string
+func createCmd(dir string, startTime time.Time, format string, name string, ext string, seq bool, seqDigits int, print bool) error {
 	if seq && format != defaultTimeFormat {
-		log.fatalErr(errors.New("The seq and format options are mutually exclusive"))
+		return errIncompatibleSeqAndFormat
 	}
+
+	var version string
+	var err error
+
+	dir = filepath.Clean(dir)
+	ext = "." + strings.TrimPrefix(ext, ".")
+
 	if seq {
-		if seqDigits <= 0 {
-			log.fatalErr(errors.New("Digits must be positive"))
-		}
-		matches, err := filepath.Glob(dir + "*" + ext)
+		matches, err := filepath.Glob(filepath.Join(dir, "*"+ext))
+
 		if err != nil {
-			log.fatalErr(err)
+			return err
 		}
-		nextSeqStr, err := nextSeq(matches, dir, seqDigits)
+
+		version, err = nextSeqVersion(matches, seqDigits)
+
 		if err != nil {
-			log.fatalErr(err)
+			return err
 		}
-		base = fmt.Sprintf("%v%v_%v.", dir, nextSeqStr, name)
 	} else {
-		switch format {
-		case "":
-			log.fatal("Time format may not be empty")
-		case "unix":
-			base = fmt.Sprintf("%v%v_%v.", dir, startTime.Unix(), name)
-		case "unixNano":
-			base = fmt.Sprintf("%v%v_%v.", dir, startTime.UnixNano(), name)
-		default:
-			base = fmt.Sprintf("%v%v_%v.", dir, startTime.Format(format), name)
+		version, err = timeVersion(startTime, format)
+
+		if err != nil {
+			return err
 		}
 	}
 
-	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
-		log.fatalErr(err)
+	versionGlob := filepath.Join(dir, version+"_*"+ext)
+	matches, err := filepath.Glob(versionGlob)
+
+	if err != nil {
+		return err
+	}
+
+	if len(matches) > 0 {
+		return fmt.Errorf("duplicate migration version: %s", version)
+	}
+
+	if err = os.MkdirAll(dir, os.ModePerm); err != nil {
+		return err
 	}
 
-	createFile(base + "up" + ext)
-	createFile(base + "down" + ext)
+	for _, direction := range []string{"up", "down"} {
+		basename := fmt.Sprintf("%s_%s.%s%s", version, name, direction, ext)
+		filename := filepath.Join(dir, basename)
+
+		if err = createFile(filename); err != nil {
+			return err
+		}
+
+		if print {
+			log.Println(filename)
+		}
+	}
+
+	return nil
 }
 
-func createFile(fname string) {
-	if _, err := os.Create(fname); err != nil {
-		log.fatalErr(err)
+func createFile(filename string) error {
+	// create exclusive (fails if file already exists)
+	f, err := os.OpenFile(filename, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
```

Comment:
Add a comment that `os.Create()` specifies `0666` as the `FileMode`, so we're doing the same